src/lib.rs

```diff
@@ -74,4 +77,10 @@ impl Evm<Db> {
 
         context.get_result()
     }
+
+    pub fn transact_and_commit(&mut self) -> Result<ExecutionResult, EVMError> {
```

Comment:
Let's rename to [`transact_commit`](https://docs.rs/revm/12.1.0/revm/struct.Evm.html#method.transact_commit)
```suggestion
    pub fn transact_commit(&mut self) -> Result<ExecutionResult, EVMError> {
```